### PR TITLE
Populate PreparsedDocumentEntry with both errors and the parsed docum…

### PR DESCRIPTION
**Pull request by zsmoore**
…ent when parsing succeeds and validation fails

See #2843 for more details

Added doc and errors in the case of successful parsing and failed validation.

Updated tests + docs

### CODE REVIEW

File: src/main/java/graphql/execution/preparsed/PreparsedDocumentEntry.java [hunk 1]
```diff
@@ -24,20 +25,24 @@ public class PreparsedDocumentEntry implements Serializable {
     private final Document document;
     private final List<? extends GraphQLError> errors;
 
-    public PreparsedDocumentEntry(Document document) {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/preparsed/PreparsedDocumentEntry.java:27]: right now  on masdter we have


```
    public PreparsedDocumentEntry(Document document) {
        assertNotNull(document);
        this.document = document;
        this.errors = null;
    }

    public PreparsedDocumentEntry(List<? extends GraphQLError> errors) {
        assertNotNull(errors);
        this.document = null;
        this.errors = errors;
    }

    public PreparsedDocumentEntry(GraphQLError error) {
        this(singletonList(assertNotNull(error)));
    }
```

This change gets rid of the `PreparsedDocumentEntry(List<? extends GraphQLError> errors)` contructor which is a SPI breaking change.  Please put it back so we can avoid an breaking change.  Sure the main code may not use it BUT any one using this today outside out code contexts will not be broken

File: src/test/groovy/graphql/execution/preparsed/PreparsedDocumentEntryTest.groovy [hunk 2]
```diff
@@ -30,34 +30,58 @@ class PreparsedDocumentEntryTest extends Specification {
...
-    def "Ensure a non-null errors returns"() {
+    def "Ensure a non-null error returns"() {
         given:
-        def errors = [new InvalidSyntaxError(new SourceLocation(0, 0), "bang"),
-                      new ValidationError(ValidationErrorType.InvalidSyntax)]
+        def error = new InvalidSyntaxError(new SourceLocation(0, 0), "bang")
 
         when:
-        def docEntry = new PreparsedDocumentEntry(errors)
+        def docEntry = new PreparsedDocumentEntry(error)
```
Comment (bbakerman) [inline src/test/groovy/graphql/execution/preparsed/PreparsedDocumentEntryTest.groovy:38]: Put this test back (as per above) and write a new one for the single error.  Again for api breaking reasons

File: src/test/groovy/graphql/language/SerialisationTest.groovy [hunk 3]
```diff
@@ -108,21 +108,39 @@ class SerialisationTest extends Specification {
...
         GraphQLError syntaxError1 = new InvalidSyntaxError(srcLoc(1, 1), "Bad Syntax 1")
-        GraphQLError validationError2 = new ValidationError(ValidationErrorType.FieldUndefined, srcLoc(2, 2), "Bad Query 2")
-        def originalEntry = new PreparsedDocumentEntry([syntaxError1, validationError2])
+        def originalEntry = new PreparsedDocumentEntry(syntaxError1)
 
         PreparsedDocumentEntry newEntry = serialisedDownAndBack(originalEntry)
 
         then:
 
-        newEntry.getErrors().size() == 2
```
Comment (bbakerman) [inline src/test/groovy/graphql/language/SerialisationTest.groovy:122]: Put this test back please - see above

**Review by bbakerman — CHANGES_REQUESTED** [review]
We dont want to make a breaking change - even if we change the way we consume the classes.  Please put back the "list of errors" contructors and the old tests and make new tests for the new constructor paths

**Comment by bbakerman** [comment]
I am otherwise happy with this change - I just dont want to break API if we can help it and we can help it here

File: src/main/java/graphql/execution/preparsed/PreparsedDocumentEntry.java (hunk 1 above)
Comment (zsmoore) [reply src/main/java/graphql/execution/preparsed/PreparsedDocumentEntry.java:27]: makes complete sense, let me add it back

File: src/test/groovy/graphql/language/SerialisationTest.groovy (hunk 3 above)
Comment (zsmoore) [reply src/test/groovy/graphql/language/SerialisationTest.groovy:122]: added back!

**Comment by zsmoore** [comment]
Thanks for the feedback @bbakerman .  Everything should be in a good state now

**Review by bbakerman — APPROVED** [review]
Thanks for this

**Pull request merged**